src/pages/QrCode/QrCode.tsx

```diff
@@ -0,0 +1,143 @@
+import { QRCodeSVG } from 'qrcode.react';
+import { TransactionCreationRequest } from '../../api/generated/payment-qr-code/TransactionCreationRequest';
+import MerchantDataForm from '../../components/MerchantDataForm';
+import TransactionDataForm from '../../components/TransactionDataForm';
+import { createTransaction, getTransaction } from '../../services/PaymentQrCodeApiService';
+import { TransactionResponse } from '../../api/generated/payment-qr-code/TransactionResponse';
+
+const initiatlState = {
+  amountCents: 0,
+  idTrxIssuer: new Date().getTime().toString(),
+  initiativeId: '63d26bbc0e71e44bb08de293',
+  mcc: '6012',
+  merchantFiscalCode: 'MERCHANT_FISCAL_CODE',
+  trxDate: new Date(),
+  vat: 'VAT',
+};
+const Home = () => {
+  const [formData, setFormData] = useState<TransactionCreationRequest>(initiatlState);
+  const [idMerchant, setIdMerchant] = useState<string>('MERCHANTID');
+  const [createTrxResponse, setCreateTrxResponse] = useState<TransactionResponse>();
+  const [statusTrx, setStatusTrx] = useState<string>();
+
+  const handleFormChange = (e: any) => {
+    const name = e.target.name;
+    const value = e.target.value;
+    setFormData((values) => ({ ...values, [name]: value }));
+  };
+
+  const handleSubmit = (_event?: any) => {
+    console.log('formData handleSUbmit', formData);
+    postCreateTransaction();
+  };
+
+  const handleReset = () => {
+    setFormData(initiatlState);
+  };
+
+  const postCreateTransaction = () => {
+    createTransaction(idMerchant, {
+      ...formData,
+      trxDate: new Date(formData.trxDate),
+    })
+      .then((res: any) => setCreateTrxResponse(res))
+      .catch((err) => console.log('err', err));
+  };
+
+  useEffect(() => {
+    if (createTrxResponse && createTrxResponse.idTrxIssuer) {
+      const interval = setInterval(() => {
+        getTransaction(idMerchant, createTrxResponse.idTrxIssuer)
+          .then((res) => {
+            // console.log('res', res);
+            setStatusTrx(res?.status);
+          })
+          .catch((_err) => clearInterval(interval));
+      }, 1000);
+
+      if (statusTrx === 'EXPECTED STATUS') {
+        return clearInterval(interval);
+      }
+    }
+  }, [JSON.stringify(createTrxResponse), statusTrx]);
+
+  return (
+    <>
+      <Box
+        sx={{
+          display: 'grid',
+          rowGap: 2,
+          pb: 2,
+          px: 2,
+        }}
+      >
+        <TitleBox
+          title="Mock portale esercente"
+          subTitle="Pagamento mediante QRCode"
+          mbTitle={2}
+          mtTitle={2}
+          mbSubTitle={5}
+          variantTitle="h4"
+          variantSubTitle="body1"
+          data-testid="title"
+        />
+        <MerchantDataForm
+          formData={formData}
+          handlechange={handleFormChange}
+          idMerchant={idMerchant}
+          setIdMerchant={setIdMerchant}
+        />
+        <TransactionDataForm formData={formData} handleChange={(e) => handleFormChange(e)} />
```

Comment:
when a method is invoked with the exact parameters, you could just provide the method's variable itself
```suggestion
        <TransactionDataForm formData={formData} handleChange={handleFormChange} />
```